### ISSUE TITLE
Nemo Extenion Unreadable icons in List View

### ISSUE THREAD

**Issue by CBtiger** [rabbitvcs/rabbitvcs#272]
The overlay icons are so small that they can hardly be seen. 

System: Xubuntu 18.04, xfce
Nemo: 4.06  (from PPA launchpad embrosyn/cinnamon)
rabbitvcs 17.01  (from master)
plugin: 17.01 (from master)
All other functions work very well. Greate to have it back to Bionic. Hopefully futureproof now with Python3. Thanks for the great work!

They icons should look like this:
[http://wiki.rabbitvcs.org/wiki/about/screenshots](url)
 
But they do look like this (can you see them?):
![RabbitVCS_nemo_SmallIcons](https://user-images.githubusercontent.com/17408411/62012504-739ed080-b187-11e9-85f5-ad470d797992.png)

Can I set the icon size somewhere? I looked around in the code but I could not finy anything. 
Any workarounds to get them bigger would be fine.

What can I provide to help closing down on the problem?

**Comment by CBtiger** [comment]
Seems that this other poster has a similr picture. that would be on cinnamon, not xfce. 
see [Post in issue #109](https://github.com/rabbitvcs/rabbitvcs/issues/251#issuecomment-515971964)

**Comment by rehierl** [comment]
I can confirm that this is what it initially looked like .. before I manually edited the emblem files.
- Nemo 4.0.6 on Linux Mint 19.1 Cinnamon
- RabbitVCS 0.17.1 (commit 5207510 from 2019-07-23)

The manual workaround I used to increase the icon sizes is as follows:
- The RabbitVCS Nemo plugin holds an array of relative image paths: e.g. `scalable/actions/rabbitvcs-ok.svg`. But these aren't the actual emblem files, which I located on my system in the following folder: `/usr/share/icons/hicolor/scalable/emblems/emblem-rabbitvcs*.svg`.
- Since these are [scalable vector graphics](https://en.wikipedia.org/wiki/Scalable_Vector_Graphics) (i.e. simple xml-based text files), you'll need a vector graphics editor (e.g. [Inkscape](https://en.wikipedia.org/wiki/Inkscape), which was used to create these files) to easily edit them.
- Basically, the original contents of these icons only fill approx 25% (1/4) of the actual icon area. So what you could do is to open the relevant 10 `emblem-rabbitvcs*.svg` files, select all its contents and manually drag them so they fill the entire icon area.
- By that you effectively rescale the icons from currently ~6% (1/16) to ~25% (1/4) of the default file/folder icon size, which is a lot better than before.

![rabbitvcs-modified-icon-sizes](https://user-images.githubusercontent.com/8310972/62823265-c026da80-bb8f-11e9-8b0a-01f9a912d27d.png)

**Comment by CBtiger** [comment]
I can confirm that it worked as you describe. The icon size in now as one would expect in xfce.

Strange that those icons where not using the full canvas size. Would there be any drawback to replace the item in the repository - like oversized icons in another distribution? 

I had hoped to find a way to add the python script with a size parameter, but I could not find where or how to do it. I saved the icons on a file server and wrote a little script to replace them during install, so this is manageable when using multiple installations.

**Comment by rehierl** [comment]
The [Icon Theme Specification](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html) mentions some `AttachPoints` that could be used to define alternative emblem positions inside `[Icon Data]` sections within the `/usr/share/icons/hicolor/index.theme` file. But these don't seem to be in use. -- There are at least none defined in my index.theme file.

Even though I didn't find a proper API documentation, I suspect that [file.add_emblem(name)](https://github.com/rabbitvcs/rabbitvcs/blob/master/clients/nemo/RabbitVCS.py#L366) is the only API method RabbitVCS can use to tell which emblem to show. Hence, I assume that Nemo/Nautilus will by default always scale emblems to fill the lower-right quadrant of a file/folder icon.

In other words, there doesn't seem to be any means to tell Nemo how to position or how to scale an emblem, which would explain why you were unable to find some sort of "size parameter". All that RabbitVCS seems to be able to do is to tell which emblem file to use. So there isn't much one can do other than to scale the actual emblems within the emblem files themselves.

But that of course doesn't explain why the emblems are by default restricted to the lower-left quadrant within the emblem files.

**Comment by CBtiger** [comment]
> Even though I didn't find a proper API documentation, I suspect that [file.add_emblem(name)](https://github.com/rabbitvcs/rabbitvcs/blob/master/clients/nemo/RabbitVCS.py#L366) is the only API method RabbitVCS can use to tell which emblem to show. Hence, I assume that Nemo/Nautilus will by default always scale emblems to fill the lower-right quadrant of a file/folder icon.
That was the bit I was looking for. The API would be nautilus/Nemo API? 
Anyway - at least that nails down the place to include any kind of scale code, should an API extension supporting this my pop up.

As there is little more to find out, I'll close this issue.
Thanks for your support.
